src/useSynthetix.js

```diff
@@ -13,7 +13,8 @@ function updateObject(oldObject, newValues) {
 const SYNTHETIX_INITIAL_STATE = Object.freeze({
   walletAddress: undefined,
   connect: () => {},
-  signature: undefined,
+  token: window.localStorage.getItem('token'),
```

Comment:
initial state should be empty. This "restore" should happen on initial load.
I also suggest storing token per wallet address so there is no problems with token mismatch